vllm/entrypoints/openai/api_server.py

```diff
@@ -1202,11 +1202,17 @@ async def invocations(raw_request: Request):
     return JSONResponse(content=res.model_dump(), status_code=res.error.code)
 
 
-if envs.VLLM_TORCH_PROFILER_DIR:
-    logger.warning(
-        "Torch Profiler is enabled in the API server. This should ONLY be "
-        "used for local development!"
-    )
+if envs.VLLM_TORCH_PROFILER_DIR or envs.VLLM_TORCH_CUDA_PROFILE:
+    if envs.VLLM_TORCH_PROFILER_DIR:
+        logger.warning(
+            "Torch Profiler is enabled in the API server. This should ONLY be "
+            "used for local development!"
+        )
+    else:
+        logger.warning(
+            "CUDA Profiler is enabled in the API server. This should ONLY be "
+            "used for local development!"
+        )
```

Comment:
nit: we should remove the outer condition e.g. `if envs.VLLM_TORCH_PROFILER_DIR:` then `elif envs.VLLM_TORCH_CUDA_PROFILE:`
also we should use warning_once